### ISSUE TITLE
[Issue] NullReferenceException in NaughtyAttributes.Editor.NaughtyEditorGUI.Dropdown (at NaughtyEditorGUI.cs:124

### ISSUE THREAD

**Issue by zephyo** [dbrizov/NaughtyAttributes#236]
Dropdown feature is really handy but basically unusable for me right now. Didn't touch the package at all, not sure why this error is happening.

Steps to reproduce:
- Use NaughtyAttributes 2.0.9 in Package Manager
- Add component whose field has the Dropdown attribute
- Get a ton of errors

![Screen Shot 2021-06-03 at 7 45 28 AM](https://user-images.githubusercontent.com/22963510/120664410-c0b4f300-c43f-11eb-88e4-0d3d1e767754.png)

**Comment by dbrizov** [comment]
Can you tell me the exact case? I am testing on my end, and everything is working. I tested with both **2.0.9** and **2.1.0**.

**Comment by zephyo** [comment]
Sorry I could've been more specific.

I found out that it only happens when I have the following:

```csharp
    [SerializeField] [Dropdown("NamespaceList")] string kataruNamespace = Namespaces.Global;
    [SerializeField] [Dropdown("CharacterList")] string reference;
    protected List<string> NamespaceList() => Namespaces.All();
    protected List<string> CharacterList() => Characters.AllInNamespace(kataruNamespace);
```

And the errors do _not_ throw when I remove the `AllInNamespace` call, so if I do either:
```csharp
    [SerializeField] [Dropdown("NamespaceList")] string kataruNamespace = Namespaces.Global;
    protected List<string> NamespaceList() => Namespaces.All();
```
or:
```csharp
    [SerializeField] [Dropdown("CharacterList")] string reference;
    protected List<string> CharacterList() => Characters.All();
```

Utility classes providing the lists:
```csharp
public static class Namespaces
    {
        public const string LilyComputer = "LilyComputer";
        public const string Global = "Global";
        public static List<string> All() => new List<string> {
          Global,
          "Prologue",
    };
}

public static class Characters
    {
        public const string None = "None",
            Lily = "Lily",
            LilyReflection = "LilyReflection",
            Dahlia = "Dahlia",
            Rose = "Rose",
            Iris = "Iris",
            Mei = "Mei",
            Think = "Think";

        public static List<string> AllInNamespace(string n)
        {
            System.Func<string, bool> predicate;
            if (n == Namespaces.Global)
            {
                predicate = r => !r.Contains(":");
            }
            else
            {
                predicate = r => r.StartsWith(n + ":");
            }
            var list = All().Where(predicate).ToList<string>();
            list.Insert(0, None);
            return list;
        }

        public static List<string> All()
        {
            var l = Jongela.Utils.Functions.PublicStaticStrings(typeof(Characters));
            var l2 = new List<string>
            {
                "BossFolder",
                "Door",
#region PROLOGUE
                "Prologue:Triggers",
                "Prologue:Spotlight",
                "Prologue:DevilArms",
#endregion
           l.AddRange(l2);
            return l;
        }
    }

```

And this only started happening once I upgraded from 2021.1.5 to 2021.1.9.

Behavior:
Upon attaching a new component with those serializable properties, a ton of those errors appear without fail and make it impossible to set the property value without it resetting to the default, but after doing other stuff in the unity editor for awhile the errors disappear and I can set the property value for real.

**Comment by zephyo** [comment]
I found a solution. The initial value of the troublesome property isn't getting set which throws the error. I change:
```csharp
 [SerializeField] [Dropdown("PassagesList")] protected string passage;
```
to:
```csharp
 [SerializeField] [Dropdown("PassagesList")] protected string passage = Passages.None;
```

**Comment by zephyo** [comment]
Yeah I saw that and thought that bug might've popped back up, so I tried initializing it. Very strange.

**Comment by dbrizov** [comment]
Fixed at 08efcae01bef12dd3f1740de6066f72718525ae5
